doc/docs/Python_Tutorials/Adjoint_Solver.md

```diff
@@ -2,13 +2,704 @@
 # Adjoint Solver
 ---
 
-Meep contains an adjoint-solver module for efficiently computing the gradient of an arbitrary function of the mode coefficients (S-parameters), DFT fields, and far fields (using the analytic near-to-far field transformation) with respect to $\varepsilon$ on a discrete spatial grid (a [`MaterialGrid`](../Python_User_Interface.md#materialgrid) class object) at multiple frequencies over a broad bandwidth. Regardless of the number of degrees of freedom for the grid points, just **two** separate timestepping runs are required. The first run is the "forward" calculation to compute the objective function and the DFT fields of the design region. The second run is the "adjoint" calculation to compute the gradient of the objective function with respect to the design variables. The adjoint run involves a special type of source distribution used to compute the DFT fields of the design region. The gradient is computed in post processing using the DFT fields from the forward and adjoint runs. This is all done automatically, and is described in:
+Meep contains an adjoint-solver module for efficiently computing the
+gradient of an arbitrary function of the mode coefficients
+($\mathcal{S}$-parameters), DFT fields, local density of states
+(LDOS), and far fields (Fraunhofer condition) with respect to
```

Comment:
The Fraunhofer condition is not required for meep's Near2Far features (regardless if performing adjoint optimization or not). Because we use a Green's function, the result is accurate at any point in space. Of course, as the point approaches the effective currents, this integral becomes more ill-conditioned... but there's no caveat on accuracy here.

If we were using a simple *Fourier transform* to perform the Near-to-far projection, then we would indeed be restricted to the "Fraunhofer regime".

Reply:
True. I have updated this line to mention this even though this point is described [elsewhere in the user manual](https://meep.readthedocs.io/en/latest/Python_User_Interface/#near-to-far-field-spectra):

```
Since the "far" fields are computed using the full Green's functions, they 
should be able to be computed anywhere outside of the near-field surface monitor. 
The only limiting factor should be discretization errors but for any given distannce, 
the "far" fields should converge to the actual DFT fields at that location with resolution 
(assuming the distance separation is >> resolution).
```

Reply:
(typo with `distannce` but otherwise LGTM)